clients/src/test/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumerTest.java

```diff
@@ -616,6 +620,80 @@ public void testCommitSyncTriggersFencedExceptionFromCommitAsync() {
         assertEquals("Get fenced exception for group.instance.id groupInstanceId1", e.getMessage());
     }
 
+    @Test
+    public void testCommitSyncAwaitsCommitAsyncCompletionWithEmptyOffsets() {
+        time = new MockTime(1);
+        consumer = newConsumer();
+
+        // Commit async (incomplete)
+        doReturn(Fetch.empty()).when(fetchCollector).collectFetch(any(FetchBuffer.class));
+        doReturn(LeaderAndEpoch.noLeaderOrEpoch()).when(metadata).currentLeader(any());
+        final TopicPartition tp = new TopicPartition("foo", 0);
+        consumer.assign(Collections.singleton(tp));
+        consumer.seek(tp, 20);
+        consumer.commitAsync();
```

Comment:
Can this be replaced with a call to `testSyncCommitTimesoutAfterIncompleteAsyncCommit()` like the other tests? I glanced back and forth a couple of times and didn't see too much difference:

```suggestion
        final TopicPartition tp = new TopicPartition("foo", 0);
        testSyncCommitTimesoutAfterIncompleteAsyncCommit(tp);
```

Reply:
I guess I may be less concerned with code duplication in test setup than my reviewers :). Done. Added another helper method that contains the lines up to the incomplete async commit.

Reply:
😆 When I see duplicate (or nearly duplicated) code, my brain turns up its sensitivity because I assume there's some devil-in-the-details reason that the code wasn't reused.

I'd assumed that it could be ever-so-slightly refactored into using `testSyncCommitTimesoutAfterIncompleteAsyncCommit()`, but no?